tests/performance/valgrind_instrumentation.h

```diff
@@ -0,0 +1,143 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016-2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_valgrind_instrumentation_h
+#define dealii_valgrind_instrumentation_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/exceptions.h>
+
+#include <unistd.h>
+#include <valgrind/callgrind.h>
+
+#include <cstdio>
+#include <fstream>
+#include <mutex>
+
+
+/**
+ * A simple wrapper around callgrind to query current instruction counter
+ * values. In order to make use of the functions in this namespace the
+ * executable has to be started with valgrind:
+ * @code
+ *   valgrind --tool=callgrind -q --combine-dumps=yes --instr-atstart=no \
+ *            --callgrind-out-file=callgrind.out ./executable
+ * @endcode
+ */
+namespace callgrind_wrapper
```

Comment:
Would you mind to rename this as follows, to stay with our usual naming scheme?
```suggestion
namespace CallgrindWrappers
```

Reply:
Changed.